source/features/index.tsx

```diff
@@ -104,12 +105,74 @@ const globalReady: Promise<RGHOptions> = new Promise(async resolve => {
 		document.head.append(<style>{options.customCSS}</style>);
 	}
 
+	Object.assign(options, await bisectFeatures());
+
 	// Create logging function
 	log = options.logging ? console.log : () => {/* No logging */};
 
 	resolve(options);
 });
 
+async function bisectFeatures(): Promise<Record<string, boolean>> {
+	const bisectedFeatures: string[] | false | undefined = await cache.get('bisect');
+	if (bisectedFeatures === undefined) {
+		return {};
+	}
+
+	if (bisectedFeatures === false) {
```

Comment:
No need for a new type
```suggestion
	if (bisectedFeatures.length === 0) {
```

Reply:
Unless I'm missing something, this is necessary to distinguish between the first step (`Every feature has been disabled. Can you see the bug?`) and the case where the user answered "Yes" to that (`Every feature has been disabled. If you still see the bug, try disabling the whole extension.`)

Reply:
If I'm reading this correctly, in the first case we're disabling all features (length === 0) while in the last step `bisect` will have no less than 2 features:

1. one enabled
2. one disabled

So at this point "yes" means 1 is buggy and "no" means 2 is buggy.

Reply:
Yes, and if there's an odd number of features it could also end up with 2 features enabled and 1 disabled, at which point if the user says no it ends by pointing the single disabled feature as the culprit.

Reply:
So we agree that `false` should be dropped, right? Just to make sure

Reply:
I think the main misunderstanding here is that `onChoiceButtonClick` always reloads the page, but instead when the user is on the last yes/no step it should just update the box without reloading.

This avoids having to set `false` altogether, because there are no more actual steps/reloads to follow.

The good part is that we already always have Exit available, so a final yes/no click only needs to:

1. Update the paragraph element
2. Drop the yes/no buttons
3. Clear the cache immediately (so that he user immediately exits the process even by closing the tab)

Reply:
> `false` should be dropped

No, I don't think so?

> 1. Remember the deselected options
> 
> 2. Deselect all
> 
> 3. Ask: can you see the bug?
>        
>   * Yes: Exit, it's not RGH
>   * No: Select half of the feature
> 
> 4. Ask: Can you see the bug?
>        
>   * Yes: Select half of the previous features
>   * No: Select half of the _other_ features
> 
> 5. Repeat 4 until there's only a feature left
> 
> 6. Tell the user which feature it is
> 
> 7. Restore previous options

We need to be able to differentiate between steps `3` and `3` + `Yes`

Reply:
I’m not sure I see the issue:

Step 3 will always receive []
- no will set [half of 1]
- yes will exit

Step 4 will always receive [half of 1], if it didn’t exit

Reply:
> shows the final message
>
> displays an Exit button

As the code is written now, this implies a reload of the page. I'll change it so it doesn't and we can drop the `false` then.